### ISSUE TITLE
[YSQL] Keyspace already exists issue caused by OID collision + X

### ISSUE THREAD

**Issue by yifanguan** [yugabyte/yugabyte-db#16441]
Jira Link: [DB-5849](https://yugabyte.atlassian.net/browse/DB-5849)
### Description

This is another side-effect issue caused by OID collision issue #16130  + X.
First,
It looks like we end up never clean up the namespace_ids_map_ after deleting a YSQL database.
In `catalog_manager.cc`
```
// Remove namespace from CatalogManager name mapping.  Will remove ID map after all Tables gone.
  {
    LockGuard lock(mutex_);
    if (namespace_names_mapper_[database->database_type()].erase(database->name()) < 1) {
      LOG(WARNING) << Format("Could not remove namespace from maps, name=$0, id=$1",
                             database->name(), database->id());
    }
  }
```

And a small repro on a fresh cluster to illustrate this issue:
```
connection 1 on node1:
\c yugabyte
CREATE DATABASE qqq; -- 16384
CREATE DATABASE ooo; -- 16385
DROP DATABASE qqq;

connection 2 on node2:
\c ooo
CREATE DATABASE ppp; -- 16384
ERROR:  Keyspace 'ppp' already exists
```
Workaround is to retry the failed create database statement.

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-5849]: https://yugabyte.atlassian.net/browse/DB-5849?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by Arjun-yb** [comment]
@yifanguan 
in **2.18.0.0-b14**, We observed 
`Failed to execute task {"sleepAfterMasterRestartMillis":180000,"sleepAfterTServerRestartMillis":180000,"nodeExporterUser":"prometheus","universeUUID":"c01f90e1-c553-496e-bc74-3572fd536dc6","enableYbc":false,"installYbc":false,"ybcInstalled":false,"encryptionAtRestConfig":{"encryptionAtRestEnabled":false,"opType":"UNDEFINED","type":"DATA_KEY"},"communicationPorts":{"masterHttpPort":7000,"masterRpcPort":7100,"tserverHttpPort":9000,"tserverRpcPort":9100,"ybControllerHttpPort":14000,"ybControllerrRpcPort":18018,"redisS..., hit error:

Task id 0ba4d5ed-0073-4b49-846b-0f7099070afd_PGSQL_TABLE_TYPE_non_colocated_db status: Failed with error COMMAND_FAILED.`
in `testysqltspcmvandbr` test
https://jenkins.dev.yugabyte.com/view/Test%20Jobs/job/itest-system-developer/5177/artifact/logs/2.18.0.0_testysqltspcmvandbr-aws-rf3_20230417_111840/testysqltspcmvandbr-aws-rf3pvlzfvfs.log
Raised issue: https://github.com/yugabyte/yugabyte-db/issues/16642
cc: @kripasreenivasan @zlareb1-yb

**Comment by yifanguan** [comment]
Resolved by commit 1c781b7
